extensions/compose/src/compose-extension.ts

```diff
@@ -0,0 +1,207 @@
+import * as path from 'node:path';
+import { existsSync, promises } from 'node:fs';
+import * as extensionApi from '@podman-desktop/api';
+import type { Detect } from './detect';
+import type { ComposeGitHubReleases } from './compose-github-releases';
+import type { OS } from './os';
+import { platform, arch } from 'node:os';
+import type { PodmanComposeGenerator } from './podman-compose-generator';
+
+export class ComposeExtension {
+  public static readonly COMPOSE_INSTALL_COMMAND = 'compose.install';
+  public static readonly COMPOSE_CHECKS_COMMAND = 'compose.checks';
+
+  public static readonly ICON_CHECK = 'fa fa-check';
+  public static readonly ICON_DOWNLOAD = 'fa fa-download';
+  public static readonly ICON_WARNING = 'fa fa-exclamation-triangle';
+
+  private statusBarItem: extensionApi.StatusBarItem | undefined;
+
+  protected currentInformation: string;
+
+  constructor(
+    private readonly extensionContext: extensionApi.ExtensionContext,
+    private readonly detect: Detect,
+    private readonly composeGitHubReleases: ComposeGitHubReleases,
+    private readonly os: OS,
+    private podmanComposeGenerator: PodmanComposeGenerator,
+  ) {}
+
+  async runChecks(firstCheck: boolean): Promise<void> {
+    this.currentInformation = undefined;
+
+    // reset status bar information
+    const statusBarChangesToApply = {
+      iconClass: '',
+      tooltip: '',
+      command: ComposeExtension.COMPOSE_CHECKS_COMMAND,
+    };
+
+    // detect if podman-compose is installed
+    const podmanComposeInstalled = await this.detect.checkForPythonPodmanCompose();
+
+    // in that case, report an issue in status bar
+    if (podmanComposeInstalled) {
+      // create a status bar item
+      statusBarChangesToApply.iconClass = ComposeExtension.ICON_WARNING;
+      this.currentInformation =
+        'This extension does not work with Python Podman Compose. It will collide with the CLI named podman-desktop. You need to uninstall Python Podman Compose before using Docker Compose v2.';
+      statusBarChangesToApply.tooltip = this.currentInformation;
+    } else {
+      // check for docker-compose
+      const dockerComposeInstalled = await this.detect.checkForDockerCompose();
+      if (dockerComposeInstalled) {
+        // check if podman-compose alias is in the PATH
+        const extensionBinFolderInPath = await this.detect.checkStoragePath();
+        if (extensionBinFolderInPath) {
+          // it's installed so we're good
+          statusBarChangesToApply.iconClass = ComposeExtension.ICON_CHECK;
+          statusBarChangesToApply.tooltip = 'Docker Compose is installed';
+        } else {
+          // not there, ask the user to setup the PATH
+          statusBarChangesToApply.iconClass = ComposeExtension.ICON_WARNING;
+          statusBarChangesToApply.tooltip = 'Path problem for Podman Compose';
+          this.currentInformation = `Please add the Podman Compose bin folder to your PATH environment variable. Value is ${path.resolve(
```

Comment:
This part is a bit confusing? It did not install for me and then tells me to do the bin folder, but the bin folder does not exist.

Reply:
code has been changed 👍